templates/microsoft_cloud_for_industry/modules/platform/variables.tf

```diff
@@ -0,0 +1,228 @@
+  type        = number
+  default     = 365
+  description = "Length of time, in days, to retain log files with usage enforced by ALZ policies."
+  validation {
+    condition     = var.log_analytics_workspace_retention_in_days >= 30 && var.log_analytics_workspace_retention_in_days <= 730
+    error_message = "The retention period must be between 30 and 730 days."
+  }
+}
+
+variable "log_analytics_solution_plans" {
+  type = list(object({
+    product   = string
+    publisher = optional(string, "Microsoft")
+  }))
+  default = [
+    {
+      product   = "OMSGallery/ContainerInsights"
+      publisher = "Microsoft"
+    },
+    {
+      product   = "OMSGallery/VMInsights"
+      publisher = "Microsoft"
+    },
+  ]
+  description = "The Log Analytics Solution Plans to create. Do not add the SecurityInsights solution plan here, this deployment method is deprecated. Instead refer to"
+  nullable    = false
+}
+
+variable "tags" {
+  type        = map(string)
+  default     = null
+  description = "(Optional) Tags that will be assigned to subscription and resources created by this deployment script."
+}
+
+variable "deploy_hub_network" {
+  type        = bool
+  default     = true
+  description = "Toggles deployment of the hub VNET. True to deploy, otherwise false. (e.g true)"
+}
+
+variable "enable_firewall" {
+  type        = bool
+  default     = true
+  description = "Toggles deployment of Azure Firewall. True to deploy, otherwise false if you don't want to change the existing firewall policies. (e.g true)"
+}
+
+variable "deploy_ddos_protection" {
+  type        = bool
+  default     = true
+  description = "Toggles deployment of Azure DDOS protection. True to deploy, otherwise false. (e.g true)"
+}
+
+variable "use_premium_firewall" {
+  type        = bool
+  default     = true
+  description = "Toggles deployment of the Premium SKU for Azure Firewall and only used if enable_Firewall is enabled. True to use Premium SKU, otherwise false. (e.g true)"
+}
+
+variable "hub_network_address_prefix" {
+  type        = string
+  default     = "10.20.0.0/16"
+  description = "CIDR range for the hub VNET. (e.g '10.20.0.0/16')|14|"
+  validation {
+    condition     = can(regex("^([0-9]{1,3}\\.){3}[0-9]{1,3}/[0-9]{1,2}$", var.hub_network_address_prefix))
+    error_message = "The hub_network_address_prefix must be a valid CIDR range, such as '10.20.0.0/16'."
+  }
+}
+
+variable "custom_subnets" {
+  type = map(object({
+    name                   = string
+    address_prefixes       = string
+    networkSecurityGroupId = optional(string, "")
+    routeTableId           = optional(string, "")
+  }))
+  default = {
+    AzureBastionSubnet = {
+      name                   = "AzureBastionSubnet"
+      address_prefixes       = "10.20.15.0/24"
+      networkSecurityGroupId = ""
+      routeTableId           = ""
+    }
+    GatewaySubnet = {
+      name                   = "GatewaySubnet"
+      address_prefixes       = "10.20.252.0/24"
```

Comment:
Gateway subnet should default to /27 or /29. /27 for ExpressRoute and / or VPN. /29 for VPN only.

Reply:
Thanks for discussing this with me. I think for now the plan is to stay with the current values we've aligned with the PMs on.